Doc page for sizeof operator incorrectly states that sizeof(T) may be used in safe code when T is constrained to be unmanaged

### Type of issue

Other (describe below)

### Description

The documentation reads:
> In safe code, the argument to the sizeof operator must be the name of an [unmanaged type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/unmanaged-types) or a type parameter that is [constrained](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters#unmanaged-constraint) to be an unmanaged type.

This is incorrect, however - the following code demonstrates that a type parameter that's constrained to be an unmanaged type may not have `sizeof` applied to it in safe code:
```
public class Foo
{
    public int GetSize<T>() where T : unmanaged
    {
        return sizeof(T);
    }
}
// error CS0233: 'T' does not have a predefined size, therefore sizeof can only be used in an unsafe context
```

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters#unmanaged-constraint correctly states, in contradiction to the above:
> The preceding method must be compiled in an unsafe context because it uses the sizeof operator on a type not known to be a built-in type.

Please can the sizeof operator doc page be corrected?

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/sizeof

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/language-reference/operators/sizeof.md

### Document Version Independent Id

7e28a662-f550-d202-9a97-e684ae1b074d

### Platform Id

86e8a1a6-0fd2-822f-93d0-d5ec8ddc72b0

### Article author

@BillWagner

### Metadata

* ID: 191acdd8-8e36-8bd4-1172-140717c36a85
* PlatformId: 86e8a1a6-0fd2-822f-93d0-d5ec8ddc72b0 
* Service: **dotnet-csharp**
* Sub-service: **lang-reference**

[Related Issues](https://github.com/dotnet/docs/issues?q=is%3Aissue+is%3Aopen+7e28a662-f550-d202-9a97-e684ae1b074d)

---
[Associated WorkItem - 537142](https://dev.azure.com/msft-skilling/Content/_workitems/edit/537142)

--- a/docs/csharp/language-reference/operators/sizeof.md
+++ b/docs/csharp/language-reference/operators/sizeof.md
@@ -1,7 +1,7 @@
 ---
 title: "sizeof operator - determine the storage needs for a type"
 description: "Learn about the C# `sizeof` operator that returns the memory amount occupied by a variable of a given type."
-ms.date: 02/06/2025
+ms.date: 12/16/2025
 f1_keywords: 
   - "sizeof_CSharpKeyword"
   - "sizeof"
@@ -10,7 +10,7 @@ helpviewer_keywords:
 ---
 # sizeof operator - determine the memory needs for a given type
 
-The `sizeof` operator returns the number of bytes occupied by a variable of a given type. In safe code, the argument to the `sizeof` operator must be the name of an [unmanaged type](../builtin-types/unmanaged-types.md) or a type parameter that is [constrained](../../programming-guide/generics/constraints-on-type-parameters.md#unmanaged-constraint) to be an unmanaged type. Unmanaged types include all numeric types, enum types, and tuple and struct types where all members are unmanaged types.
+The `sizeof` operator returns the number of bytes occupied by a variable of a given type. In safe code, the argument to the `sizeof` operator must be the name of a builtin [unmanaged type](../builtin-types/unmanaged-types.md).
 
 The expressions presented in the following table are evaluated in compile time to the corresponding constant values and don't require an unsafe context:
 
@@ -30,15 +30,18 @@ The expressions presented in the following table are evaluated in compile time t
 | `sizeof(decimal)` | 16             |
 | `sizeof(bool)`    |  1             |
 
-In unsafe code, the argument to `sizeof` can include pointer types and managed types, including unconstrained type parameters. Examples include `object` and `string`.
+The size of a builtin unmanaged type is a compile time constant.
+
+In [unsafe](../keywords/unsafe.md) code you can use `sizeof` as follows:
+
+- A type parameter that is constrained](../../programming-guide/generics/constraints-on-type-parameters.md#unmanaged-constraint) to be an unmanaged type returns the size of that unmanaged type at runtime.
+- A managed type or a pointer type returns the size of the reference or pointer, not the size of the object it refers to.
 
 The following example demonstrates the usage of the `sizeof` operator:
 
 :::code language="csharp" source="./snippets/shared/SizeOfOperator.cs":::
 
 The `sizeof` operator returns the number of bytes allocated by the common language runtime in managed memory. For [struct](../builtin-types/struct.md) types, that value includes any padding, as the preceding example demonstrates. The result of the `sizeof` operator might differ from the result of the <xref:System.Runtime.InteropServices.Marshal.SizeOf%2A?displayProperty=nameWithType> method, which returns the size of a type in *unmanaged* memory.
-
-In unsafe code, when the argument is a managed type, the `sizeof` operator returns the size of a reference, not the number of bytes allocated for an instance of that type.
 
 > [!IMPORTANT]
 >

--- a/docs/csharp/language-reference/operators/sizeof.md
+++ b/docs/csharp/language-reference/operators/sizeof.md
@@ -30,7 +30,7 @@ The expressions presented in the following table are evaluated at compile time t
 | `sizeof(decimal)` | 16             |
 | `sizeof(bool)`    |  1             |
 
-The size of a built-in unmanaged type is a compile-time constant.
+The size of a built-in, unmanaged type is a compile-time constant.
 
 In [unsafe](../keywords/unsafe.md) code, you can use `sizeof` as follows:
 

--- a/docs/csharp/language-reference/operators/sizeof.md
+++ b/docs/csharp/language-reference/operators/sizeof.md
@@ -10,9 +10,9 @@ helpviewer_keywords:
 ---
 # sizeof operator - determine the memory needs for a given type
 
-The `sizeof` operator returns the number of bytes occupied by a variable of a given type. In safe code, the argument to the `sizeof` operator must be the name of a builtin [unmanaged type](../builtin-types/unmanaged-types.md).
+The `sizeof` operator returns the number of bytes occupied by a variable of a given type. In safe code, the argument to the `sizeof` operator must be the name of a built-in [unmanaged type](../builtin-types/unmanaged-types.md).
 
-The expressions presented in the following table are evaluated in compile time to the corresponding constant values and don't require an unsafe context:
+The expressions presented in the following table are evaluated at compile time to the corresponding constant values and don't require an unsafe context:
 
 | Expression        | Constant value |
 |-------------------|----------------|
@@ -30,11 +30,11 @@ The expressions presented in the following table are evaluated in compile time t
 | `sizeof(decimal)` | 16             |
 | `sizeof(bool)`    |  1             |
 
-The size of a builtin unmanaged type is a compile time constant.
+The size of a built-in unmanaged type is a compile-time constant.
 
-In [unsafe](../keywords/unsafe.md) code you can use `sizeof` as follows:
+In [unsafe](../keywords/unsafe.md) code, you can use `sizeof` as follows:
 
-- A type parameter that is constrained](../../programming-guide/generics/constraints-on-type-parameters.md#unmanaged-constraint) to be an unmanaged type returns the size of that unmanaged type at runtime.
+- A type parameter that is [constrained](../../programming-guide/generics/constraints-on-type-parameters.md#unmanaged-constraint) to be an unmanaged type returns the size of that unmanaged type at runtime.
 - A managed type or a pointer type returns the size of the reference or pointer, not the size of the object it refers to.
 
 The following example demonstrates the usage of the `sizeof` operator:

